### ISSUE TITLE
crossplane xpkg push results in 2 packages pushed and one has "--" appended to it

### ISSUE THREAD

**Issue by Chrand123** [crossplane/crossplane#4980]
### What happened?

When I ran crossplane xpkg push on a package/image I built using crossplane xpkg build, instead of 1 image being pushed to the common repository, 2 artifacts got pushed. One was a normal docker image (with the docker logo) and the other was another type of file with a logo that had the letters "Ci" in a box. Also, the docker image didn't get pushed with the same exact name as what I specified when pushing it. Instead, it had "--" appended to it. 

### How can we reproduce it?

First create a Composition YAML file, a Configuration YAML file, and a CompositeResourceDefinition YAML. Fill them out with very basic arbitrary details needed to build the package. Then run crossplane xpkg build. Then take the xpkg that was built, and push it to a common repository (I used JFrog). Then, check what the pushed image/pkg looks like in the common repository. 

### What environment did it happen in?
Crossplane version: v1.14.0

More details
* Cloud provider - Azure
* Kubernetes version - v5.0.1
* OS - Mac

**Comment by mateusz-lubanski-sinch** [comment]
not sure if it is related but when I used new CLI to build and publish Configuration package, after deploying it on k8s cluster I am getting below error:

```
Warning  ParsePackage  42s (x12 over 5m39s)  packages/configurationrevision.pkg.crossplane.io  cannot initialize parser  backend: failed to fetch package from remote: no child with platform linux/amd64 in index nexus.int.clxnetworks.net:8089/.../topic/configuration:v0.0.1
```

**Comment by jbw976** [comment]
Yes @mateusz-lubanski-sinch, that definitely looks related.  #4977 should fix this issue, which has been backported to the v1.14 branch and should be released in a v1.14.1 patch in the next few days 🙏

**Comment by jbw976** [comment]
The fix for this issue should be included in the v1.14.1 release: https://github.com/crossplane/crossplane/releases/tag/v1.14.1
